src/main/java/uk/gov/companieshouse/limitedpartnershipsapi/service/LimitedPartnerValidator.java

```diff
@@ -0,0 +1,157 @@
+package uk.gov.companieshouse.limitedpartnershipsapi.service;
+
+import jakarta.validation.ConstraintViolation;
+import jakarta.validation.Validator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.MethodParameter;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.BeanPropertyBindingResult;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import uk.gov.companieshouse.api.model.validationstatus.ValidationStatusError;
+import uk.gov.companieshouse.limitedpartnershipsapi.exception.ServiceException;
+import uk.gov.companieshouse.limitedpartnershipsapi.model.limitedpartner.dto.LimitedPartnerDataDto;
+import uk.gov.companieshouse.limitedpartnershipsapi.model.limitedpartner.dto.LimitedPartnerDto;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+@Component
+public class LimitedPartnerValidator {
+
+    @Autowired
+    private Validator validator;
+
+
+    public List<ValidationStatusError> validate(LimitedPartnerDto limitedPartnerDto) throws ServiceException {
+        List<ValidationStatusError> errorsList = new ArrayList<>();
+
+        executeJavaBeansValidation(limitedPartnerDto, errorsList);
+
+        var dataDto = limitedPartnerDto.getData();
+        if (dataDto.isLegalEntity()) {
+            if (dataDto.getPrincipalOfficeAddress() == null) {
+                errorsList.add(createValidationStatusError("Principal office address is required", LimitedPartnerDataDto.PRINCIPAL_OFFICE_ADDRESS_FIELD));
+            }
+        } else {
+            if (dataDto.getUsualResidentialAddress() == null) {
+                errorsList.add(createValidationStatusError("Usual residential address is required", LimitedPartnerDataDto.USUAL_RESIDENTIAL_ADDRESS_FIELD));
+            }
+        }
+
+        return errorsList;
+    }
+
+    public void isValid(LimitedPartnerDto limitedPartnerDto) throws NoSuchMethodException, MethodArgumentNotValidException {
```

Comment:
i wouldn't call this method isValid as 'is' methods should return a boolean. 'validate' would be better.